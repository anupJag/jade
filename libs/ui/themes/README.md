# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
