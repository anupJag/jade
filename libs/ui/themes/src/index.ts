export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
