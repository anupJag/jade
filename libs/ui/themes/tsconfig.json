{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
