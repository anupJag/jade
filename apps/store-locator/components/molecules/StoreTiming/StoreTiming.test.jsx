import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from '@jade/ui/themes';

import { StoreTiming } from './index';

describe('StoreTiming', () => {
  let StoreTimingComponent = '';
  let timing = {};
  let info = '';

  beforeEach(() => {
    timing = {
      mon: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      tue: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      wed: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      thu: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      fri: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      sat: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      sun: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
    };

    info = 'Opening hours vary for some facilities.';

    StoreTimingComponent = render(
      <ThemeProvider>
        <StoreTiming timing={timing} info={info} />
      </ThemeProvider>,
    );
  });

  it('should render store timing correctly', async () => {
    const { asFragment } = StoreTimingComponent;
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have info text', () => {
    expect(screen.getByText(info).textContent).toEqual('Opening hours vary for some facilities.');
  });
});
