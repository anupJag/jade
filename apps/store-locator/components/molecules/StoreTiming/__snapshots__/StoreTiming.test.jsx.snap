// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreTiming should render store timing correctly 1`] = `
<DocumentFragment>
  <div
    class="css-f051v4"
  >
    <div
      class="css-k0f1kz"
    />
    <div
      class="css-0"
    >
      <h4
        class="css-mcy282"
      >
        <p
          class="css-1t24val"
        >
            
        </p>
      </h4>
      <div
        class="css-2o23i1"
      >
        <div
          class="css-gg4vpm"
        >
          <span>
            Mon
          </span>
          <span>
            07:00 - 23:00
          </span>
        </div>
      </div>
      <div
        class="css-2o23i1"
      >
        <div
          class="css-gg4vpm"
        >
          <span>
            Tue
          </span>
          <span>
            07:00 - 23:00
          </span>
        </div>
      </div>
      <div
        class="css-2o23i1"
      >
        <div
          class="css-gg4vpm"
        >
          <span>
            Wed
          </span>
          <span>
            07:00 - 23:00
          </span>
        </div>
      </div>
      <div
        class="css-2o23i1"
      >
        <div
          class="css-gg4vpm"
        >
          <span>
            Thu
          </span>
          <span>
            07:00 - 23:00
          </span>
        </div>
      </div>
      <div
        class="css-2o23i1"
      >
        <div
          class="css-gg4vpm"
        >
          <span>
            Fri
          </span>
          <span>
            07:00 - 23:00
          </span>
        </div>
      </div>
      <div
        class="css-2o23i1"
      >
        <div
          class="css-gg4vpm"
        >
          <span>
            Sat
          </span>
          <span>
            07:00 - 23:00
          </span>
        </div>
      </div>
      <div
        class="css-2o23i1"
      >
        <div
          class="css-gg4vpm"
        >
          <span>
            Sun
          </span>
          <span>
            07:00 - 23:00
          </span>
        </div>
      </div>
    </div>
    <div
      class="css-1n1nucz"
    >
      Opening hours vary for some facilities.
    </div>
  </div>
</DocumentFragment>
`;
